webextension/privileged/ui/cloudstorage.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* global ExtensionAPI */
+
+/**
+ * Handles the Cloud Downloads UI in download panel.
+ */
+
+"use strict";
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+
+this.cloudstorage = class extends ExtensionAPI {
+  getAPI(context) {
+    const { CloudDownloadsView } = ChromeUtils.import(
+      context.extension.rootURI.resolve("privileged/ui/CloudDownloadsView.jsm")
+    );
+
+    function cleanUpPrefs() {
+      // Ensure cloud storage study related prefs are cleared
+      const CLOUD_SERVICES_PREF = "cloud.services.";
+      Services.prefs.clearUserPref(CLOUD_SERVICES_PREF + "lastprompt");
+      Services.prefs.clearUserPref(CLOUD_SERVICES_PREF + "interval.prompt");
+      Services.prefs.clearUserPref(CLOUD_SERVICES_PREF + "storage.key");
+      Services.prefs.clearUserPref(CLOUD_SERVICES_PREF + "api.enabled");
+    }
+
+    context.extension.callOnClose({
+        close: () => {
+          if (context.extension.shutdownReason == "ADDON_UNINSTALL" ||
+              context.extension.shutdownReason == "ADDON_DISABLE") {
+            Services.prefs.setBoolPref("cloud.services.api.enabled", false);
+            cleanUpPrefs();
+          }
+        },
+    });
+
+    return {
+      cloudstorage: {
+        async init(path) {
+          CloudDownloadsView.stylesURL = path;
+
+          let isAPIEnabled = Services.prefs.getBoolPref("cloud.services.api.enabled", false);
+          if (isAPIEnabled) {
+            await CloudDownloadsView.init();
+          } else {
+            let interval = 0; // TBD: Interval to be picked from Study Utils Config
+            Services.prefs.setBoolPref("cloud.services.api.enabled", true);
+            Services.prefs.setCharPref("cloud.services.interval.prompt", interval);
+          }
+          return path;
+        },
+
+        async uninit() {
```

Comment:
Not sure if this needs to be `async`. I don't see an `await` or returning a Promise. Plus, I'm not sure if `Services.prefs.setBoolPref()` is asynchronous or not.